jena-arq/Grammar/arq.jj

```diff
@@ -500,6 +498,27 @@ void QuadsNotTriples(QuadAccSink acc) : {Node gn ; Node prev = acc.getGraph() ;
     <RBRACE>
     { setAccGraph(acc, prev) ; }
 }
+void ConstructQuads(QuadAcc acc) : {}
+{
+    (TriplesTemplate(acc))?
+    (
+       ConstructQuadsNotTriples(acc)
+       (<DOT>)?
+       (TriplesTemplate(acc))?
+    )*
+}
+void ConstructQuadsNotTriples(QuadAccSink acc) : {Node gn = org.apache.jena.sparql.core.Quad.defaultGraphNodeGenerated ; Node prev = acc.getGraph() ; }
+{
```

Comment:
No need to write the package name - add an import then this code can be shorter.